test/distributed/test_ddp_with_rpc.py

```diff
@@ -0,0 +1,444 @@
+            )
+            gLogger.info(f"Running distributed optimizer...")
+            dist_optimizer = DistributedOptimizer(
+                optim.SGD,
+                self.get_real_module().remote_parameters_rrefs
+                + [
+                    rpc.RRef(local_param)
+                    for local_param in self.hybrid_module.parameters()
+                ],
+                lr=LR,
+            )
+            dist_optimizer.step(context_id)
+
+            self.print_parameters()
+
+        gLogger.info(f"Loss: {loss}.")
+        return loss
+
+
+def get_training_examples():
+    n = 8
+    training_examples = FeatureSet(
+        dense_features=torch.zeros((n, D_DENSE)),
+        sparse_features=torch.zeros(n, dtype=torch.long),
+        labels=torch.zeros(n, dtype=torch.long),
+    )
+    idx = 0
+    for x in range(2):
+        for y in range(2):
+            for z in range(2):
+                training_examples.dense_features[idx, :] = torch.Tensor((x, y))
+                training_examples.sparse_features[idx] = z
+                training_examples.labels[idx] = x ^ y ^ z
+                idx += 1
+    # Split the examples among NUM_TRAINERS trainers
+    examples_per_trainer = int(n / NUM_TRAINERS)
+    return [
+        FeatureSet(
+            dense_features=training_examples.dense_features[
+                start : start + examples_per_trainer, :
+            ],
+            sparse_features=training_examples.sparse_features[
+                start : start + examples_per_trainer
+            ],
+            labels=training_examples.labels[
+                start : start + examples_per_trainer
+            ],
+        )
+        for start in range(0, n, examples_per_trainer)
+    ]
+
+
+class TestDdpWithRpc(MultiProcessTestCase):
+    rpc_backend = rpc.backend_registry.BackendType.PROCESS_GROUP
+    rpc_backend_options = None
+
+    @property
+    def world_size(self) -> int:
+        return WORLD_SIZE
+
+    def remote_worker_name(self) -> str:
+        # The name has to be consistent with that in 'dist_init' decorator.
+        return f"worker{REMOTE_WORKER_RANK}"
+
+    def trainer_name(self, rank):
+        # The name has to be consistent with that in 'dist_init' decorator.
+        return f"worker{rank}"
+
+    def setUp(self):
+        super(TestDdpWithRpc, self).setUp()
+
+        os.environ["MASTER_ADDR"] = str(MASTER_ADDR)
+        os.environ["MASTER_PORT"] = str(MASTER_PORT)
+        self._spawn_processes()
+
+    def tearDown(self):
+        super(TestDdpWithRpc, self).tearDown()
+
+    @dist_init
+    def _remote_worker_process(self):
+        process_group_for_ddp = dist_c10d.new_group(ranks=TRAINER_RANKS)
+        gLogger.info(f"The remote worker is running.")
+        dist.destroy_process_group(process_group_for_ddp)
+        gLogger.info(f"Exiting remote worker.")
+
+    @dist_init
+    def _trainer_process(self, rank: int):
+        gLogger.info(f"Running the trainer #{rank}...")
```

Comment:
Same as above, wouldn't trainer process exit before we have a chance to run some computations?